### PR TITLE
feat: monitor page backgrounding events and close streaming connection

### PR DESCRIPTION
**Pull request by ajwootto**

### CODE REVIEW

**Comment by ajwootto** [comment]
Current dependencies on/for this PR:
* main
  * **PR #280** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/280" target="_blank">Graphite</a> 
    * **PR #281** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/281" target="_blank">Graphite</a> 
      * **PR #282** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/282" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/282?utm_source=stack-comment).

**Review by nsalamad — APPROVED** [review]

File: sdk/js/src/Client.ts [hunk 1]
```diff
@@ -271,6 +273,25 @@ export class DVCClient implements Client {
...
+                        this.logger.error('Failed to refetch config', e)
+                    })
+                    this.streamingConnection?.reopen()
+                } else {
+                    this.logger.debug('Page is not visible, closing streaming connection')
+                    this.streamingConnection?.close()
+                }
+            }
+
+            document.addEventListener?.('visibilitychange', this.pageVisibilityHandler)
```
Comment (jsalaber) [inline sdk/js/src/Client.ts:292]: should we also remove this event listener when the page closes?

**Review by jsalaber — CHANGES_REQUESTED** [review]

File: sdk/js/src/Client.ts (hunk 1 above)
Comment (ajwootto) [reply sdk/js/src/Client.ts:292]: you mean like when the page is actually closed? there can't be any event handlers after the page has closed anyway

Comment (nsalamad) [reply sdk/js/src/Client.ts:292]: he does that in the next PR lol

**Review by jsalaber — APPROVED** [review]

**Pull request merged**